Name is now readonly

The name field is not readonly when posting new ads

[ERROR] Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/admin/Development/kleinanzeigen-bot/kleinanzeigen_bot/__main__.py", line 8, in <module>
    kleinanzeigen_bot.main(sys.argv)
  File "/home/admin/Development/kleinanzeigen-bot/kleinanzeigen_bot/__init__.py", line 1014, in main
    KleinanzeigenBot().run(args)
  File "/home/admin/Development/kleinanzeigen-bot/kleinanzeigen_bot/__init__.py", line 80, in run
    self.publish_ads(ads)
  File "/home/admin/Development/kleinanzeigen-bot/kleinanzeigen_bot/__init__.py", line 444, in publish_ads
    self.publish_ad(ad_file, ad_cfg, ad_cfg_orig)
  File "/home/admin/Development/kleinanzeigen-bot/kleinanzeigen_bot/__init__.py", line 552, in publish_ad
    self.web_input(By.ID, "postad-contactname", ad_cfg["contact"]["name"])
  File "/home/admin/Development/kleinanzeigen-bot/kleinanzeigen_bot/selenium_mixin.py", line 305, in web_input
    input_field.clear()
  File "/home/admin/Development/kleinanzeigen-bot/.venv/lib/python3.11/site-packages/selenium/webdriver/remote/webelement.py", line 117, in clear
    self._execute(Command.CLEAR_ELEMENT)
  File "/home/admin/Development/kleinanzeigen-bot/.venv/lib/python3.11/site-packages/selenium/webdriver/remote/webelement.py", line 395, in _execute
    return self._parent.execute(command, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Development/kleinanzeigen-bot/.venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 346, in execute
    self.error_handler.check_response(response)
  File "/home/admin/Development/kleinanzeigen-bot/.venv/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.InvalidElementStateException: Message: invalid element state
  (Session info: chrome=118.0.5993.88)
Stacktrace:
#0 0x55810b366d34 <unknown>
#1 0x55810b078df7 <unknown>
#2 0x55810b0b7197 <unknown>
#3 0x55810b0e0892 <unknown>
#4 0x55810b0b4ea1 <unknown>
#5 0x55810b0e0b0e <unknown>
#6 0x55810b0f767c <unknown>
#7 0x55810b0e0633 <unknown>
#8 0x55810b0b2eb7 <unknown>
#9 0x55810b0b4513 <unknown>
#10 0x55810b339c47 <unknown>
#11 0x55810b33ceb0 <unknown>
#12 0x55810b33c928 <unknown>
#13 0x55810b33d3c5 <unknown>
#14 0x55810b34409b <unknown>
#15 0x55810b33d792 <unknown>
#16 0x55810b318370 <unknown>
#17 0x55810b3574c7 <unknown>
#18 0x55810b3576d5 <unknown>
#19 0x55810b36633e <unknown>
#20 0x7fe3b7dbd9eb <unknown>

--- a/kleinanzeigen_bot/__init__.py
+++ b/kleinanzeigen_bot/__init__.py
@@ -505,7 +505,7 @@ class KleinanzeigenBot(SeleniumMixin):
             self.__set_shipping_options(ad_cfg)
         else:
             try:
-                self.web_click(By.XPATH, '//*[contains(@class, "SelectionButton--Right")]')
+                self.web_click(By.XPATH, '//*[contains(@class, "jsx-2623555103")]')
                 self.web_click(By.XPATH, '//*[contains(@class, "CarrierSelectionModal--Button")]')
                 self.web_click(By.XPATH, '//*[contains(@class, "CarrierOption--Main")]')
                 if ad_cfg["shipping_costs"]:
@@ -532,10 +532,10 @@ class KleinanzeigenBot(SeleniumMixin):
         sell_directly = ad_cfg["sell_directly"]
         try:
             if sell_directly and ad_cfg["shipping_type"] == "SHIPPING" and ad_cfg["shipping_options"] and price_type in {"FIXED", "NEGOTIABLE"}:
-                if not self.webdriver.find_element(By.ID, "buy-now-toggle").is_selected():
-                    self.web_click(By.XPATH, '//*[contains(@class, "BuyNowSection")]//div[contains(@class, "Toggle--Slider")]')
-            elif self.webdriver.find_element(By.ID, "buy-now-toggle").is_selected():
-                self.web_click(By.XPATH, '//*[contains(@class, "BuyNowSection")]//span[contains(@class, "Toggle--Slider")]')
+                if not self.webdriver.find_element(By.ID, "radio-buy-now-yes").is_selected():
+                    self.web_click(By.XPATH, '//*[contains(@id, "radio-buy-now-yes")]')
+            elif not self.webdriver.find_element(By.ID, "radio-buy-now-no").is_selected():
+                self.web_click(By.XPATH, '//*[contains(@id, "radio-buy-now-no")]')
         except NoSuchElementException as ex:
             LOG.debug(ex, exc_info = True)
 
@@ -566,7 +566,7 @@ class KleinanzeigenBot(SeleniumMixin):
         #############################
         # set contact name
         #############################
-        if ad_cfg["contact"]["name"]:
+        if ad_cfg["contact"]["name"] and not self.webdriver.find_element(By.ID, "postad-contactname").get_attribute("readonly"):
             self.web_input(By.ID, "postad-contactname", ad_cfg["contact"]["name"])
 
         #############################

